platform/backend/repository/memory/src/main/kotlin/io/hamal/repository/memory/record/Code.kt

```diff
@@ -88,13 +88,15 @@ class MemoryCodeRepository : MemoryRecordRepository<CodeId, CodeRecord, Code>(
                 versionOf(codeId, cmd.id)
             } else {
                 val currentVersion = versionOf(codeId, cmd.id)
-                store(
-                    CodeUpdatedRecord(
-                        entityId = codeId,
-                        cmdId = cmd.id,
-                        value = cmd.value ?: currentVersion.value
+                if (cmd.value != currentVersion.value && cmd.value != null) {
+                    store(
+                        CodeUpdatedRecord(
+                            entityId = codeId,
+                            cmdId = cmd.id,
+                            value = cmd.value!!
+                        )
```

Comment:
Help the kotlin compiler to do some smart casting

```suggestion
             val codeValue = cmd.value
                if (codeValue != null && codeValue != currentVersion.value ) {
                    store(
                        CodeUpdatedRecord(
                            entityId = codeId,
                            cmdId = cmd.id,
                            value = codeValue
                        )
```

Reply:
As a convention do null checks always first!